packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+		const node = this.getNode()!;
+
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: `Interviewing ${this.ccName}...`,
+			direction: "none",
+		});
+
+		await this.refreshValues();
+
+		// Remember that the interview is complete
+		this.interviewComplete = true;
+	}
+
+	public async refreshValues(): Promise<void> {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses["Door Lock Logging"].withOptions({
+			priority: MessagePriority.NodeQuery,
+		});
+
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: "querying supported records number...",
+			direction: "outbound",
+		});
+		const supportedRecordsNumber = await api.getSupportedRecordsNumber();
+
+		if (!supportedRecordsNumber) {
+			this.driver.controllerLog.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message:
+					"Door Lock Logging supported records number query timed out, skipping interview...",
+				level: "warn",
+			});
+			return;
+		}
+
+		const supportedNumberLogMessage = `received response for supported records number: ${supportedRecordsNumber}`;
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: supportedNumberLogMessage,
+			direction: "inbound",
+		});
+
+		setRecordMetadata.call(this, supportedRecordsNumber);
+
+		// Just retrieve the lastest record
+		const record = await api.getRecord(LATEST_RECORD_NUMBER_KEY);
+
+		if (!record) {
+			this.driver.controllerLog.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message:
+					"Door Lock Logging record query timed out, skipping interview...",
+				level: "warn",
+			});
+			return;
+		}
+
+		const recordLogMessage = `received response for latest record: ${JSON.stringify(
+			record,
+		)}`;
+		this.driver.controllerLog.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: recordLogMessage,
+			direction: "inbound",
+		});
+	}
+}
+
+@CCCommand(DoorLockLoggingCommand.RecordsSupportedReport)
+export class DoorLockLoggingCCRecordsSupportedReport extends DoorLockLoggingCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		validatePayload(this.payload.length >= 1);
+
+		this.supportedRecordsNumber = this.payload[0];
+		this.persistValues();
+	}
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Max logging records stored",
+	})
+	public readonly supportedRecordsNumber: number;
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"supported records number": this.supportedRecordsNumber,
+			},
+		};
+	}
+
+	public persistValues(): boolean {
+		if (!super.persistValues()) return false;
+
+		this.getValueDB().setValue(
+			getDoorLockLoggingRecordsSupportedValueID(this.endpointIndex),
+			this.supportedRecordsNumber,
+		);
+
+		return true;
+	}
```

Comment:
Remove this, values marked with @ccValue and @ccValueMetadata are persisted automatically